modules/network/src/main/java/org/apache/ignite/internal/network/recovery/RecoveryClientHandshakeManager.java

```diff
@@ -354,6 +364,22 @@ private void handleClusterIdMismatch(HandshakeStartMessage msg) {
         sendRejectionMessageAndFailHandshake(message, HandshakeRejectionReason.CLUSTER_ID_MISMATCH, HandshakeException::new);
     }
 
+    private void handleProductNameMismatch(HandshakeStartMessage msg) {
+        String message = msg.serverNode().name() + ":" + msg.serverNode().id()
```

Comment:
I think the code won't be hot and you can use a formatted string to make it more visual. It's up to you.